lib/spack/docs/build_systems/bundlepackage.rst

```diff
@@ -9,9 +9,20 @@
 Bundle
 ------
 
-``BundlePackage`` represents a set of packages that are expected to work well
-together, such as a collection of commonly used software libraries.  The
-associated software is specified as bundle dependencies.
+``BundlePackage`` represents a set of packages that are expected to work
+well together, such as a collection of commonly used software libraries.
+The associated software is specified as bundle dependencies.
+
+If it makes sense, variants and conflicts can be added to the package.
+Variants could ensure that common build options are consistent across
+the packages supporting them. Conflicts are used to prevent builds with
+known bugs or issues. For example, if the bundle is known to only build
+on ``linux`` then conflicts can be added, such as:
+
+.. code-block:: python
+
+    for platform in ["cray", "darwin", "windows"]:
+        conflicts(f"platform={platform}", msg="Only builds on linux")
```

Comment:
why not:

```python
    require("platform=linux")
```

also some links to bundle packages (like xSDK or the vis SDK) might be useful

Reply:
I was trying to illustrate the use of the for-loop above the directive so maybe that isn't the best example.

Great point about the links.  Thanks.

Reply:
Not sure that the for loop really matters for the point here (though you'll get it across if you just link to existing sdk's -- they have for loops).

If you want to show require and conflict, you could show require for linux, and maybe a `conflicts("platform=windows")` for something that only works on unix